### PR TITLE
validation output changes

### PR DESCRIPTION
**Pull request by najiyacl**
Screenshot 2023-07-13 at 11 56 17 AM
Screenshot 2023-07-13 at 11 56 58 AM

Intergration Test - https://teamcity.oci.oraclecorp.com/buildConfiguration/Odsc_FeatureStore_FeatureStoreAdsIntegrationTests_FeatureStoreAdsIntegrationTestsPython38/40693002?hideProblemsFromDependencies=false&hideTestsFromDependencies=false

### CODE REVIEW

File: ads/feature_store/validation_output.py [hunk 1]
```diff
@@ -22,9 +23,29 @@ def to_pandas(self) -> pd.DataFrame:
...
         """
         if self.content:
-            profile_result = pd.json_normalize(self.content).transpose()
+            validation_output_json = (
+                json.loads(self.content)
+            )
+            profile_result = pd.json_normalize(validation_output_json.get("results")).transpose()
             return profile_result
 
+    def to_summary(self) -> pd.DataFrame:
```
Comment (KshitizLohia) [inline ads/feature_store/validation_output.py:32]: this looks much better. i know not a part of this jira but is it possible to also change to_pandas for stats and remove `{"value": xyz}` and just have `xyz`

File: ads/feature_store/validation_output.py [hunk 2]
```diff
@@ -22,9 +23,29 @@ def to_pandas(self) -> pd.DataFrame:
...
+        """
+        Converts the validation output summary information to a pandas DataFrame.
+
+        Returns
+        -------
+        pd.DataFrame
+            The validation output summary information as a pandas DataFrame.
+        """
+        if self.content:
```
Comment (KshitizLohia) [inline ads/feature_store/validation_output.py:41]: we need to update documentation and have a release for 1.2 version of feature store in release notes with all the changes clubbed
1. [FEATURE] `to_summary()` for validation
2. [FEATURE] `to_pandas()` for validation
3. [FEATURE] `model.save()` with dataset to model mapping
4. [FEATURE] `conda` release if that is done
5. [FIX] fixing unit test action for github to bring the pr in mergeable state

Comment (najiyacl) [reply ads/feature_store/validation_output.py:41]: 1,2,8,9 will update the doc as part of this PR.
3.will be done in the model PR
4.conda will be updated after the ADS build 
5.This I am thinking we will have en env variable to get the url and use it in the client,Once the authorizer change to get the endpoint is done ,we can incorporate that, but still will keep the env variable for the user to provide the endpoint manually.
7.Are there UT failures?

File: ads/feature_store/docs/source/dataset.rst [hunk 3]
```diff
@@ -119,28 +119,56 @@ With a Dataset instance, we can get the last dataset job details using ``get_las
...
 
+.. note::
+
+  Great Expectations is a Python-based open-source library for validating, documenting, and profiling your data. It helps you to maintain data quality and improve communication about data between teams. Software developers have long known that automated testing is essential for managing complex codebases.
+
+.. image:: figures/validation.png
+
 .. code-block:: python3
 
-  feature_group.save_expectation(expectation_suite, expectation_type="STRICT")
```
Comment (KshitizLohia) [inline ads/feature_store/docs/source/dataset.rst:143]: Is there a problem with the indentation in this section?

File: ads/feature_store/docs/source/feature_group.rst [hunk 4]
```diff
@@ -152,22 +152,55 @@ Feature store provides an API similar to Pandas to join feature groups together
...
+         )
+    )
+
+You can call the ``get_validation_output()`` method of the FeatureGroup instance to fetch validation results for a specific ingestion job.
+The ``get_validation_output()`` method takes the following optional parameter:
+
+- ``job_id: string``. Id of feature group job
+``get_validation_output().to_pandas()`` will output  the validation results for each expectation as pandas dataframe
+
+.. image:: figures/validation_results.png
```
Comment (KshitizLohia) [inline ads/feature_store/docs/source/feature_group.rst:199]: we need to add image for validation_results.png and validation_summary.png in the pr. also, this is a manual step and we need to upload image to ociodscdev manually

File: ads/feature_store/docs/source/dataset.rst [hunk 5]
```diff
@@ -119,28 +119,56 @@ With a Dataset instance, we can get the last dataset job details using ``get_las
...
+                expectation_type=ExpectationType.STRICT,
+            )
+        )
+
+You can call the ``get_validation_output()`` method of the Dataset instance to fetch validation results for a specific ingestion job.
+The ``get_validation_output()`` method takes the following optional parameter:
+
+- ``job_id: string``. Id of dataset job
+
+``get_validation_output().to_pandas()`` will output  the validation results for each expectation as pandas dataframe
```
Comment (KshitizLohia) [inline ads/feature_store/docs/source/dataset.rst:169]: I think that the section lacks the inclusion of the following images: `figures/validation_results.png` and `figures/validation_summary.png`.

File: ads/feature_store/docs/source/feature_group.rst (hunk 4 above)
Comment (najiyacl) [reply ads/feature_store/docs/source/feature_group.rst:199]: added the images to the PR

File: ads/feature_store/docs/source/dataset.rst [hunk 6]
```diff
@@ -119,28 +119,56 @@ With a Dataset instance, we can get the last dataset job details using ``get_las
...
+            .with_entity_id(entity_id)
+            .with_feature_store_id(feature_store_id)
+            .with_query(f"SELECT * FROM `{entity_id}`.{feature_group_name}")
+            .with_expectation_suite(
+                expectation_suite=expectation_suite,
+                expectation_type=ExpectationType.STRICT,
+            )
+        )
+
+You can call the ``get_validation_output()`` method of the Dataset instance to fetch validation results for a specific ingestion job.
```
Comment (KshitizLohia) [inline ads/feature_store/docs/source/dataset.rst:164]: i think it appears necessary to consider updating the information in `validation.rst` by incorporating these details, and also removing the `Expectation Suite` section from the document in `validation.rst`.

File: ads/feature_store/docs/source/dataset.rst (hunk 3 above)
Comment (najiyacl) [reply ads/feature_store/docs/source/dataset.rst:143]: Screenshot 2023-07-17 at 11 45 39 AM

File: ads/feature_store/docs/source/feature_group.rst [hunk 7]
```diff
@@ -152,22 +152,55 @@ Feature store provides an API similar to Pandas to join feature groups together
 
 Save expectation entity
 =======================
-Feature store allows you to define expectations on data being materialized into feature group instance. With a ``FeatureGroup`` instance, we can save the expectation entity using ``save_expectation()``
```
Comment (KshitizLohia) [inline ads/feature_store/docs/source/feature_group.rst:155]: we can consider incorporating modifications in the upcoming 1.2 release notes by updating `release_notes.rst`, and other can contribute to the 1.2 release notes until we identify an approach to combine multiple releases.

File: ads/feature_store/docs/source/dataset.rst (hunk 5 above)
Comment (najiyacl) [reply ads/feature_store/docs/source/dataset.rst:169]: added new images in dataset

File: ads/feature_store/docs/source/feature_group.rst (hunk 7 above)
Comment (najiyacl) [reply ads/feature_store/docs/source/feature_group.rst:155]: Should I update now or once we have all planned features?

Comment (KshitizLohia) [reply ads/feature_store/docs/source/feature_group.rst:155]: you can update for your set of changes and others can add on top of it?

**Review by KshitizLohia — APPROVED** [review]

**Pull request merged**